apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ApolloCompiler.kt

```diff
@@ -93,6 +93,10 @@ object ApolloCompiler {
 
     findApolloReservedEnumValueNames(schema).checkNoErrors()
 
+    if (!options.flattenModels) {
+      findCapitalizedFields(definitions + incomingFragments).checkNoErrors()
```

Comment:
No need to check the incomingFragments, they'll be checked by upstream modules:

```suggestion
      findCapitalizedFields(definitions).checkNoErrors()
```